raiden/ui/cli.py

```diff
@@ -1025,18 +1039,28 @@ def _run_smoketest():
     elif args['transport'] == 'matrix' and local_matrix.lower() != 'none':
         print('WARNING: The Matrix transport is experimental')
         args['mapped_socket'] = None
-        with HTTPExecutor(
+        print_step('starting matrix transport')
+        try:
+            with HTTPExecutor(
                 local_matrix,
                 status=r'^[24]\d\d$',
-                url=urljoin(args['matrix_server'], '/_matrix'),
-        ):
-            args['extra_config'] = {
-                'matrix': {
-                    'discovery_room': {'server': 'matrix.local.raiden'},
-                    'server_name': 'matrix.local.raiden',
-                },
-            }
-            success = _run_smoketest()
+                url=urljoin(args['matrix_server'], '/_matrix/client/versions'),
+                shell=True,
+            ):
+                args['extra_config'] = {
+                    'matrix': {
+                        'discovery_room': {'server': 'matrix.local.raiden'},
+                        'server_name': 'matrix.local.raiden',
+                    },
+                }
+                success = _run_smoketest()
+        except PermissionError:
+            append_report('matrix server start exception', traceback.format_exc())
+            print_step(
```

Comment:
this introduced an extra step, that doesnt have a counterpart for the happy case, then the counter will be wrong if an exception is raised

Reply:
It doesn't add a step since the exception will be triggered before calling `_run_smoketest` so it will actually finish at step `[6/8]`, but I guess that's fine for an error condition.